src/btf_parser.h

```diff
@@ -3,13 +3,171 @@
 
 #pragma once
 #include <functional>
+#include <iostream>
 #include <stdint.h>
 #include <string>
+#include <variant>
 #include <vector>
 
 typedef std::function<void(const std::string& section, uint32_t instruction_offset, const std::string& file_name,
                            const std::string& source, uint32_t line_number, uint32_t column_number)>
     btf_line_info_visitor;
+
+typedef uint32_t btf_type_id;
+
+typedef struct btf_kind_int {
```

Comment:
Why the typedef? It's useful for C compatibility, but this file can't be compatible with C code.